man/fi_psm2.7.md

```diff
@@ -199,6 +199,23 @@ The *psm2* provider checks for the following environment variables:
 
   The default setting is 2.
 
+*FI_PSM2_LAZY_CONN*
+: There are two strategies on when to establish connections between the PSM2
+  endpoints that OFI endpoints are built on top of. In eager connection mode,
+  connections are established when addresses are inserted into the address
+  vector. In lazy connection mode, connections are established when addresses
+  are used the first time in communication. Eager connection mode has slightly
+  lower critical path overhead but lazy connection mode scales better.
+
+  This option controls how the two connection modes are used. When set to 1,
+  lazy connection mode is always used. When set to 0, eager connection mode
+  is used when required conditions are all met and lazy connection mode is
+  used otherwise. The conditions to used eager connection mode are: (1) only
+  one endpoint is opened by each process; (2) scalable endpoint is not used;
+  and (3) the address vector type is FI_AV_MAP.
```

Comment:
So, eager connections are the default in most cases?

Reply:
Condition (2) should read "scalable endpoint is disabled". This is true for Intel MPI non-"mt" build because it explicitly sets PSM2_MULTI_EP=0.

For applications that doesn't set PSM2_MULTI_EP, the provider would set PSM2_MULTI_EP=1 and thus dynamic connection will always be used.